segment_vector.go

```diff
@@ -64,3 +68,11 @@ type VecPosting interface {
 
 	Size() int
 }
+
+type VectorIndex interface {
+	// Searches for the top 'k' similar vectors to the ones in 'x'
+	// Returns the scores and IDs for the similar vectors.
+	Search(x []float32, k int64) (distances []float32, labels []int64, err error)
```

Comment:
`scores and IDs` in the comment but `distances` and `labels` as the return parameter names, keep things consistent shall we?